SharpRepository.Repository/RepositoryBase.cs

```diff
@@ -33,7 +33,7 @@ protected RepositoryBase(ICachingStrategy<T, TKey> cachingStrategy = null)
             _aspects = entityType.GetAllAttributes<RepositoryActionBaseAttribute>(inherit: true)
                 .Where(x => x.Enabled)
                 .OrderBy(x => x.Order)
-                .ToArray();
+                .ToDictionary(a => a.GetType().FullName, a => a);
```

Comment:
Just noticed 1 other thing that I think is a problem.  Storing the items in a Dictionary means that they aren't ordered anymore, where they were in the list.  So now the call to `_aspects.Values.Where(a => a.Enabled)` in the `RunAspect` method won't be in the order that the attributes define.

We will need to add an ordering to the `RunAspect` below in order to maintain that.